lighthouse-core/gather/gatherers/dobetterweb/tags-blocking-first-paint.js

```diff
@@ -20,162 +19,184 @@
+        if (tag.tagName !== 'LINK') return false;
+
+        // Filter stylesheet/HTML imports that block rendering.
+        // https://www.igvita.com/2012/06/14/debunking-responsive-css-performance-myths/
+        // https://www.w3.org/TR/html-imports/#dfn-import-async-attribute
+        const linkTag = /** @type {HTMLLinkElement} */ (tag);
+        const blockingStylesheet = linkTag.rel === 'stylesheet' &&
+          window.matchMedia(linkTag.media).matches && !linkTag.disabled;
+        const blockingImport = linkTag.rel === 'import' && !linkTag.hasAttribute('async');
+        return blockingStylesheet || blockingImport;
+      })
+      .map(tag => {
+        return {
+          tagName: /** @type {'LINK'} */ ('LINK'),
+          url: tag.href,
+          href: tag.href,
+          rel: tag.rel,
+          media: tag.media,
+          disabled: tag.disabled,
+          mediaChanges: linkMediaChanges.filter(item => item.href === tag.href),
+        };
+      });
+
+    const scriptTags = [...document.querySelectorAll('head script[src]')]
+      .filter(/** @return {tag is HTMLScriptElement} */ tag => {
+        if (tag.tagName !== 'SCRIPT') return false;
+
+        const scriptTag = /** @type {HTMLScriptElement} */ (tag);
+        return (
+          !scriptTag.hasAttribute('async') &&
+          !scriptTag.hasAttribute('defer') &&
+          !/^data:/.test(scriptTag.src) &&
+          !/^blob:/.test(scriptTag.src) &&
+          scriptTag.getAttribute('type') !== 'module'
+        );
+      })
+      .map(tag => {
+        return {
+          tagName: /** @type {'SCRIPT'} */ ('SCRIPT'),
+          url: tag.src,
+          src: tag.src,
+        };
+      });
+
+    return [...linkTags, ...scriptTags];
+  } catch (e) {
+    const friendly = 'Unable to gather Scripts/Stylesheets/HTML Imports on the page';
+    throw new Error(`${friendly}: ${e.message}`);
+  }
 }
 
 class TagsBlockingFirstPaint extends Gatherer {
   /**
    * @param {Array<LH.Artifacts.NetworkRequest>} networkRecords
+   * @return {Map<string, LH.Artifacts.NetworkRequest>}
    */
   static _filteredAndIndexedByUrl(networkRecords) {
-    /** @type {Object<string, {isLinkPreload: boolean, transferSize: number, startTime: number, endTime: number}>} */
-    const result = {};
+    /** @type {Map<string, LH.Artifacts.NetworkRequest>} */
+    const result = new Map();
 
-    return networkRecords.reduce((prev, record) => {
-      if (!record.finished) {
-        return prev;
-      }
+    for (const record of networkRecords) {
+      if (!record.finished) continue;
 
       const isParserGenerated = record.initiator.type === 'parser';
       // A stylesheet only blocks script if it was initiated by the parser
       // https://html.spec.whatwg.org/multipage/semantics.html#interactions-of-styling-and-scripting
       const isParserScriptOrStyle = /(css|script)/.test(record.mimeType) && isParserGenerated;
-      const isFailedRequest = record._failed;
```

Comment:
this is an outright bug (there hasn't been a `NetworkRequest._failed` in a long time), part of the danger of `// @ts-nocheck`. It's not clear how consequential it will be. /cc @patrickhulce

Reply:
Oh nice find! it was more of an edge case so I wouldn't expect it to affect that many sites.